// @porsche-design-system/components-js/ag-grid/esm/styles.mjs
/* Auto Generated Start */
const radiusLg = "8px";
const borderWidthThin = "1px";
const colorSuccessFrostedSoft = "light-dark(hsl(109 80% 95% / 0.55),hsl(157 59% 15% / 0.66))";
const radiusMd = "6px";
const colorSuccessLow = "light-dark(hsl(115 77.5% 27.8% / 0.18),hsl(157 84.9% 41.6% / 0.18))";
const radiusSm = "4px";
const radiusXl = "12px";
const colorCanvas = "light-dark(#fff,hsl(225 66.7% 1.2%))";
const colorContrastLower = "light-dark(hsl(234 6% 32.9% / 0.324),hsl(240 1.5% 61.8% / 0.302))";
const colorContrastLow = "light-dark(hsl(240 5.3% 14.9% / 0.5),hsl(240 12.5% 96.9% / 0.45))";
const colorErrorLow = "light-dark(hsl(357 78% 41% / 0.18),hsl(0 96.9% 62% / 0.18))";
const colorFocus = "light-dark(#1A44EA,#1A44EA)";
const colorFrosted = "light-dark(hsl(240 5% 70% / 0.148),hsl(240 2% 43% / 0.228))";
const colorFrostedSoft = "light-dark(hsl(234 9.8% 60% / 0.06),hsl(240 3.7% 26.5% / 0.154))";
const colorInfo = "light-dark(hsl(228 83.2% 51%),hsl(210 100% 54.5%))";
const colorInfoLow = "light-dark(hsl(228 83.2% 51% / 0.18),hsl(210 100% 54.5% / 0.18))";
const colorPrimary = "light-dark(hsl(225 66.7% 1.2%),hsl(225 100% 99%))";
const colorSuccess = "light-dark(hsl(115 77.5% 27.8%),hsl(157 84.9% 41.6%))";
const colorSurface = "light-dark(hsl(240 10% 95%),hsl(240 2% 10%))";
const fontPorscheNext = "'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif";
const typescaleSm = "1rem";
const typescaleXs = ".875rem";
const fontWeightSemibold = 600;
const spacingStaticXs = "4px";
/* Auto Generated End */

export { borderWidthThin, colorCanvas, colorContrastLow, colorContrastLower, colorErrorLow, colorFocus, colorFrosted, colorFrostedSoft, colorInfo, colorInfoLow, colorPrimary, colorSuccess, colorSuccessFrostedSoft, colorSuccessLow, colorSurface, fontPorscheNext, fontWeightSemibold, radiusLg, radiusMd, radiusSm, radiusXl, spacingStaticXs, typescaleSm, typescaleXs };
