// @porsche-design-system/components-js/ag-grid/esm/parts/input-styles.mjs
import { createPart } from 'ag-grid-community';
import { borderWidthThin, colorContrastLower, colorCanvas, colorPrimary, colorErrorLow, radiusXl } from '../styles.mjs';

/**
 * Input field styles for AG Grid following v35 standards
 *
 * Provides custom styling for input elements with Porsche Design System colors.
 * Supports both light and dark theme via CSS `color-scheme`.
 *
 * Includes styling for:
 * - Normal state
 * - Focus state
 * - Invalid state
 * - Disabled state
 */
const inputStyles = createPart({
    feature: 'inputStyles',
    params: {
        // Normal state
        inputBorder: `solid ${borderWidthThin} ${colorContrastLower}`,
        inputDisabledBackgroundColor: colorCanvas,
        // Focus state
        inputFocusBorder: `solid ${borderWidthThin} ${colorPrimary}`,
        // Invalid state
        inputInvalidBorder: `solid ${borderWidthThin} ${colorErrorLow}`,
        // Disabled state
        inputDisabledBorder: `solid ${borderWidthThin} light-dark(hsla(233,6.6%,23.9%,0.412),hsla(240,1.5%,61.8%,0.302))`,
    },
    css: `
    /* Custom border radius for input fields */
    .ag-text-field-input {
        border-radius: ${radiusXl};
    }
  `,
});

export { inputStyles };
